src/tools/miri/tests/fail/dangling_pointers/storage_dead_dangling.stderr

```diff
@@ -1,8 +1,8 @@
 error: Undefined Behavior: dereferencing pointer failed: $HEX[noalloc] is a dangling pointer (it has no provenance)
   --> $DIR/storage_dead_dangling.rs:LL:CC
    |
-LL |     unsafe { &mut *(LEAK as *mut i32) };
-   |              ^^^^^^^^^^^^^^^^^^^^^^^^ dereferencing pointer failed: $HEX[noalloc] is a dangling pointer (it has no provenance)
+LL |     let _ = unsafe { &mut *(LEAK as *mut i32) };
```

Comment:
FWIW, adding `let _` here is not always obviously the right choice: there are cases where `*y;` is UB but `let _ = *y;` is fine. Using `*y;` as a statement is equivalent to `let _unused = *y;`, not to `let _ = *y;`.

However, only "fail" tests were changed in Miri, so if UB was accidentally removed we would have noticed.